scripts/deploy/create-image.sh

```diff
@@ -0,0 +1,137 @@
+
+# remove any backup repo
+echo "removing backup repo"
+rm -rf $COMPANION_HOME/.companion || error "failed to remove backup repo"
+
+# keep log files for future debug
+#rm -rf $COMPANION_HOME/.webui.log
+#rm -rf $COMPANION_HOME/.update_log
+
+# set default service parameters
+echo "copying gstreamer2.param.default"
+cp ../../params/gstreamer2.param.default $COMPANION_HOME/gstreamer2.param || error "failed to copy gstreamer2.param.default"
+echo "copying mavproxy.param.default"
+cp ../../params/mavproxy.param.default $COMPANION_HOME/mavproxy.param || error "failed to copy mavproxy.param.default"
+
+# remove any wifi information
+echo "copying wpa_suplicant.conf.default"
+cp ../../params/wpa_supplicant.conf.default $MOUNT_LOCATION/etc/wpa_supplicant/wpa_supplicant.conf || error "failed to copy default wpa_supplicant.conf"
+
+# insert expand_fs command in /etc/rc.local
+# above the line to start the companion services
+echo "adding expand_fs entry to /etc/rc.local"
+EXPAND="/home/pi/companion/scripts/expand_fs.sh"
+sed -i "\%$EXPAND%d" $MOUNT_LOCATION/etc/rc.local || error "sed failed to remove expand_fs entry in /etc/rc.local"
+sed -i "\%^. /home/pi/companion/.companion.rc%i$EXPAND" $MOUNT_LOCATION/etc/rc.local || error "sed failed to add expand_fs entry in /etc/rc.local"
+
+umount $MOUNT_LOCATION
+
+# check filesystem
+e2fsck -n -f $DEV_PART2 || error "e2fsck failed to verify filesystem"
+
+# resize2fs will not run unless e2fsck was run with *only* the -f option
+# so we run it again, without the -n (no user input) option
+# we assume it won't fail because it did not the first time
+e2fsck -f $DEV_PART2 || error "e2fsck failed to verify filesystem"
+
+# shrink filesystem to minimum size
+resize2fs -M $DEV_PART2 || error "resize2fs failed to resize partition"
+
+# find new end of filesystem
+dumpe2fs $DEV_PART2 -h
+BLOCK_COUNT=$(dumpe2fs $DEV_PART2 -h | grep "Block count" | sed 's/[^0-9]*//g')
+BLOCK_SIZE=$(dumpe2fs $DEV_PART2 -h | grep "Block size" | sed 's/[^0-9]*//g')
+PARTITION_SIZE_BYTES=$(($BLOCK_SIZE * $BLOCK_COUNT))
+BEGIN_DATA=$(parted -m $DEV_DISK unit b print | grep ^2: | cut -d: -f2 | tr -d B)
+NEW_END=$(($BEGIN_DATA + $PARTITION_SIZE_BYTES))
+
+# shrink partition to match filesystem
+# parted -s $DEV_DISK unit b resizepart 2 $NEW_END
+
+# the above command does not work because there is a bug with the --script option
+# this workaround was proposed a the thread about the bug here
+# https://bugs.launchpad.net/ubuntu/+source/parted/+bug/1270203
+echo -e "yes\n" | parted ---pretend-input-tty $DEV_DISK unit b resizepart 2 $NEW_END || error "parted failed to shrink partition"
+
+END_DATA=$(parted -m $DEV_DISK unit MiB print | grep ^2: | cut -d: -f3 | cut -f1 -dM)
+
+echo "The end of the data is $END_DATA MiB"
+echo "Do you want to copy $END_DATA MiB from $DEV_DISK to $IMAGE_FILE?"
+input "y"
+
+dd if=$DEV_DISK of=$IMAGE_FILE bs=1MiB count=$END_DATA status=progress || error "Failed to copy $DEV_DISK to $IMAGE_FILE"
```

Comment:
maybe add a `sync` around here? I'm not quite sure if that is necessary.

Reply:
done